angularjs-portal-home/src/main/webapp/my-app/layout/widget/partials/weather.html

```diff
@@ -1,23 +1,23 @@
 <div>
 
-    <div ng-if="!weather.location">
-		<div ng-model="noLocations" ng-init="noLocations=true">
-			<i class="fa fa-exclamation-triangle" aria-label='Warning'></i>
-			<div class="warning-message-weather-widget">Hold up! You haven't set any weather locations yet! Launch the full app to select your weather locations.</div>
-		</div>
-		
-		<div ng-if="!noLocations"id="loading">
-    		<loading-gif data-object='weatherData'></loading-gif>
-    	</div>
-	</div>
+    <div ng-if="weatherData && weatherData.length == 0 && !loading && !error">
+        <div>
+            <i class="fa fa-exclamation-triangle" aria-label='Warning'></i>
+            <div class="warning-message-weather-widget">Hold up! You haven't set any weather locations yet! Launch the full app to select your weather locations.</div>
+        </div>
+    </div>
+    
+    <div ng-if="loading" id="loading">
+       <loading-gif data-object='weatherData'></loading-gif>
+   </div>
 
-    <div ng-if="weather.location && weatherData.length == 0">
-	    <i class="fa fa-frown-o" aria-label="Error"></i>
-        <div class="error-message-weather-widget"> Oops! The weather service seems to be down. <a href="http://www.weather.com/weather/today/l/USWI0411:1:US" target="_blank">Try clicking here for Madison weather!</a></div>
-	</div>
-	
+    <div ng-if="error">
+        <i class="fa fa-frown-o" aria-label="Error"></i>
+        <p class="credit"> Oops! The weather service seems to be down. <a href="http://www.weather.com/weather/today/l/USWI0411:1:US" target="_blank">Try clicking here for Madison weather!</a></p>
```

Comment:
Not a deal breaker, but `credit` as the class for an error message doesn't feel totally clean.